File: java/src/main/java/org/nitk/hackathon/account/Savings.java
package org.nitk.hackathon.account;

public class Savings implements Account {

    private String accountNumber;
    private double balance;

    public Savings(String accountNumber) {
        this.accountNumber = accountNumber;
        this.balance = 0;
    }

    public String getAccountNumber() {
        return accountNumber;
    }

    @Override
    public void deposit(double amount) {
        balance += amount;
    }

    @Override
    public void withdraw(double amount) {
        balance -= amount;
    }

    @Override
    public double getBalance() {
        return balance;
    }

}

File: python/account/savings.py

from python.account.account import Account


class Savings(Account):
    """
    A concrete implementation of Account for Savings account.
    """

    def deposit(self, amount):
        """
        To deposit a specified amount into the savings account
        """
        self.balance += amount

    def withdraw(self, amount):
        """
        To withdraw a specified amount from the savings account
        """
        if self.balance >= amount:
            self.balance -= amount
        else:
            raise ValueError("Insufficient funds")

    def get_balance(self):
        """
        Method to get the savings account balance
        """
        return self.balance

    def get_account_number(self):
        """
        Method to get the savings account number
        """
        return self.account_number